fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java

```diff
@@ -270,25 +275,52 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
 
     public void alterMaterializedViewStatus(MaterializedView materializedView, String status, boolean isReplay) {
         if (AlterMaterializedViewStatusClause.ACTIVE.equalsIgnoreCase(status)) {
-            String viewDefineSql = materializedView.getViewDefineSql();
             ConnectContext context = new ConnectContext();
+            Optional<Database> mayDb = GlobalStateMgr.getCurrentState().mayGetDb(materializedView.getDbId());
+            context.setGlobalStateMgr(GlobalStateMgr.getCurrentState());
             context.setQualifiedUser(AuthenticationMgr.ROOT_USER);
             context.setCurrentUserIdentity(UserIdentity.ROOT);
             context.setCurrentRoleIds(Sets.newHashSet(PrivilegeBuiltinConstants.ROOT_ROLE_ID));
 
-            List<StatementBase> statementBaseList = SqlParser.parse(viewDefineSql, context.getSessionVariable());
-            QueryStatement queryStatement = (QueryStatement) statementBaseList.get(0);
+            // If we could parse the MV sql successfully, and the schema of mv does not change,
+            // we could reuse the existing MV
+            String createMvSql = materializedView.getMaterializedViewDdlStmt(false);
+            QueryStatement mvQueryStatement = null;
             try {
-                Analyzer.analyze(queryStatement, context);
+                // check database existing
+                String dbName = mayDb.orElseThrow(() ->
```

Comment:
Could we extract the validate logic to a new method?  which should be more clear. we should avoid large method.

Reply:
done